renku/api/models/run.py

```diff
@@ -58,17 +59,30 @@
 )
 from renku.core.plugins.provider import RENKU_ENV_PREFIX
 
+name_pattern = re.compile("[a-zA-Z0-9-_]+")
+
 
 class _PathBase(PathLike):
     @ensure_project_context
-    def __init__(self, path, project):
-        self._path = Path(path)
+    def __init__(self, name: str, path: Union[str, Path], project):
+        if name and not name_pattern.match(name):
+            raise errors.ParameterError(
+                f"Name {name} contains illegal characters. Only characters, numbers, _ and - are allowed."
+            )
+        self.name = name
+
+        env_value = environ.get(f"{RENKU_ENV_PREFIX}{name}", None)
+
+        if env_value:
+            self._path = Path(env_value)
+        else:
+            self._path = Path(path)
 
-        indirect_list_path = self._get_indirect_list_path(project.path)
-        indirect_list_path.parent.mkdir(exist_ok=True, parents=True)
-        with open(indirect_list_path, "a") as file_:
-            file_.write(str(path))
-            file_.write("\n")
+            indirect_list_path = self._get_indirect_list_path(project.path)
+            indirect_list_path.parent.mkdir(exist_ok=True, parents=True)
+            with open(indirect_list_path, "a") as file_:
+                file_.write(f"{path}:::{name}")
```

Comment:
I'd rather go with a common format (e.g. JSON) instead of this. The original use case for the indirect in/out was that users should write some paths to these files and then Renku reads and adds them to in/outs. So, users need to know about this format and it would be easier if the format is familiar.